### ISSUE TITLE
Recursive flows

### ISSUE THREAD

**Issue by iceman3000** [microsoft/promptflow#2279]
**Is your feature request related to a problem? Please describe.**
Inability to have recursion in flows.

**Describe the solution you'd like**
Ability to have a loop within the flow

**Describe alternatives you've considered**
Build recursion into a python node and complete all logic in that.

**Additional context**
Especially with the introduction of tool calling, and more "agent like" uses of OpenAI, it would be handy to be able to (based on logic), repeat a section of the flow. 

e.g. Customer asks a question, search query is generated, request is made to docs, docs are reviewed, new search query is generated based off learning from the docs, docs are reviewed, then response is given to the customer (if AI is now happy with answer).

**Comment by D-W-** [comment]
Hi @iceman3000 , we don't have plan to introduce recursion/loops in DAG flow since it's not easy to write it in YAML. But we'll introduce a flexible way to execute code script with prompt flow, where your can define orchestration logic with python code. Will that meet your need?

**Comment by D-W-** [comment]
Hi @LastRemote , in my understanding, in we model above process as a flow, it can be implemented with flex mode:

```python
def entry():
    while need_more_information(question, result):
        auto_selection_result = auto_select(question)
        exp_selection_result = explain_select(question)
        result = retrive(auto_selection_result, exp_selection_result)
```

Will this meet your requirements?

**Comment by ElliotWood** [comment]
> Hi @LastRemote , in my understanding, in we model above process as a flow, it can be implemented with flex mode:
> 
> ```python
> def entry():
>     while need_more_information(question, result):
>         auto_selection_result = auto_select(question)
>         exp_selection_result = explain_select(question)
>         result = retrive(auto_selection_result, exp_selection_result)
> ```
> 
> Will this meet your requirements?

Could we please see a complete example of this - also a description of 'flex mode'?

**Comment by D-W-** [comment]
> > Hi @LastRemote , in my understanding, in we model above process as a flow, it can be implemented with flex mode:
> > ```python
> > def entry():
> >     while need_more_information(question, result):
> >         auto_selection_result = auto_select(question)
> >         exp_selection_result = explain_select(question)
> >         result = retrive(auto_selection_result, exp_selection_result)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Will this meet your requirements?
> 
> Could we please see a complete example of this - also a description of 'flex mode'?

Hi @ElliotWood , flex mode is a mode we're working on to directly execute code script with prompt flow with tracing ability. The code script will be provided by user so user can run complex orchestration logic in it. (e.g. the while loop in above example). Since it's still working in progress, we currently don't have examples/docs about it. I'll update here when it's publicly available.

**Comment by GuilhermeAbacherli** [comment]
Hello guys, first thanks for this nice lib.
Upvoted on the request too.

I am using version `1.17.0`.

@D-W-  is there any news on how to create a recursive Flow?

I managed to create flow controls (recursion) using Python nodes, but the only point where I got stuck is how to request LLM tools again from this "parallel flow" (that I am controlling via Python node).

The problem, for me, is that LLM nodes can only be executed once after a trigger from an input update. After the LLM node is invoked for the first time, it is marked as done, and I don't know how to invoke it again, then the whole flow ends (after the first round of execution).

I would appreciate any heklp on this. Thanks in advance!
